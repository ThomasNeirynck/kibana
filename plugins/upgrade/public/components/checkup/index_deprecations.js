import _ from 'lodash';
import React from 'react';

import { StatusGroup } from '../status_group';
import { Issues } from './issues';
import { IssueSummary } from './issue_summary';


export function IndexDeprecations({ className, deprecations }) {
  const indexNames = Object.keys(deprecations);
  const levelCountsByIndex = _.reduce(deprecations, (result, levelCounts, indexName) => ({
    ...result,
    [indexName]: _.countBy(levelCounts, 'level'),
  }), {});
  const totalLevelCounts = _.reduce(levelCountsByIndex, (totalCounts, levelCounts) => ({
    none: totalCounts.none + (levelCounts.none || 0),
    info: totalCounts.info + (levelCounts.info || 0),
    warning: totalCounts.warning + (levelCounts.warning || 0),
    critical: totalCounts.critical + (levelCounts.critical || 0),
  }), {
    none: 0,
    info: 0,
    warning: 0,
    critical: 0,
  });

  return (
    <StatusGroup
      className={ className }
      isInitiallyCollapsed
      status={ <IssueSummary issueLevelCounts={ totalLevelCounts } /> }
      title="Index Settings"
    >
      { indexNames.length > 0
        ? (
          indexNames.map((indexName) => (
            <div key={ indexName }>
              <div className="kuiBar">
                <div className="kuiBarSection">
                  <p className='kuiTextTitle'>{ indexName }</p>
                </div>
                <div className="kuiBarSection">
                  <IssueSummary issueLevelCounts={ levelCountsByIndex[indexName] } />
                </div>
              </div>
              <Issues issues={ deprecations[indexName] } />
            </div>
          ))
        )
        : <p className="kuiText kuiSubduedText">No index-level deprecations</p>
      }
    </StatusGroup>
  );
}

IndexDeprecations.propTypes = {
  className: React.PropTypes.string,
  deprecations: React.PropTypes.objectOf(React.PropTypes.array),
};

IndexDeprecations.defaultProps = {
  className: null,
  deprecations: {},
};
