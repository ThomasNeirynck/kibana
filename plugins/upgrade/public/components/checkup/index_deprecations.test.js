import { shallow } from 'enzyme';
import React from 'react';

import { IndexDeprecations } from './index_deprecations';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('IndexDeprecations', () => {
  test('renders the className property', () => {
    const component = (
      <IndexDeprecations className="myClass" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a success message when there are no deprecations', () => {
    const component = (
      <IndexDeprecations deprecations={ {} }/>
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders Issues and IssueSummary components for each index when there are deprecations', () => {
    const deprecations = {
      index1: [
        { level: 'info', },
        { level: 'critical', },
        { level: 'critical', },
      ],
      index2: [
        { level: 'warning', },
        { level: 'info', },
        { level: 'none', },
      ],
    };
    const component = (
      <IndexDeprecations deprecations={ deprecations }/>
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
