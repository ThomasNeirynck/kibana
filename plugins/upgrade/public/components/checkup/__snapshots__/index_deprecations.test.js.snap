// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexDeprecations renders Issues and IssueSummary components for each index when there are deprecations 1`] = `
<StatusGroup
  className={null}
  isInitiallyCollapsed={true}
  status={
    <IssueSummary
      className={null}
      issueLevelCounts={
        Object {
          "critical": 2,
          "info": 2,
          "none": 1,
          "warning": 1,
        }
      }
    />
  }
  title="Index Settings"
>
  <div>
    <div
      className="kuiBar"
    >
      <div
        className="kuiBarSection"
      >
        <p
          className="kuiTextTitle"
        >
          index1
        </p>
      </div>
      <div
        className="kuiBarSection"
      >
        <IssueSummary
          className={null}
          issueLevelCounts={
            Object {
              "critical": 2,
              "info": 1,
            }
          }
        />
      </div>
    </div>
    <Issues
      className={null}
      issues={
        Array [
          Object {
            "level": "info",
          },
          Object {
            "level": "critical",
          },
          Object {
            "level": "critical",
          },
        ]
      }
    />
  </div>
  <div>
    <div
      className="kuiBar"
    >
      <div
        className="kuiBarSection"
      >
        <p
          className="kuiTextTitle"
        >
          index2
        </p>
      </div>
      <div
        className="kuiBarSection"
      >
        <IssueSummary
          className={null}
          issueLevelCounts={
            Object {
              "info": 1,
              "none": 1,
              "warning": 1,
            }
          }
        />
      </div>
    </div>
    <Issues
      className={null}
      issues={
        Array [
          Object {
            "level": "warning",
          },
          Object {
            "level": "info",
          },
          Object {
            "level": "none",
          },
        ]
      }
    />
  </div>
</StatusGroup>
`;

exports[`IndexDeprecations renders a success message when there are no deprecations 1`] = `
<StatusGroup
  className={null}
  isInitiallyCollapsed={true}
  status={
    <IssueSummary
      className={null}
      issueLevelCounts={
        Object {
          "critical": 0,
          "info": 0,
          "none": 0,
          "warning": 0,
        }
      }
    />
  }
  title="Index Settings"
>
  <p
    className="kuiText kuiSubduedText"
  >
    No index-level deprecations
  </p>
</StatusGroup>
`;

exports[`IndexDeprecations renders the className property 1`] = `
<StatusGroup
  className="myClass"
  isInitiallyCollapsed={true}
  status={
    <IssueSummary
      className={null}
      issueLevelCounts={
        Object {
          "critical": 0,
          "info": 0,
          "none": 0,
          "warning": 0,
        }
      }
    />
  }
  title="Index Settings"
>
  <p
    className="kuiText kuiSubduedText"
  >
    No index-level deprecations
  </p>
</StatusGroup>
`;
